eox_nelp/pearson_vue/exceptions.py

```diff
@@ -0,0 +1,47 @@
+"""
+Module to add managed exceptions related Pearson Vue Integration
+"""
+
+
+class PearsonBaseError(Exception):
+    """Pearson Base error class
+    Most classes that inherit from this class must have exception_type.
+    """
+    exception_type = "base-error"
+
+    def __init__(self, *args):
+        self.exception_type = self.exception_type
+        super().__init__(self.exception_type, *args)
+
```

Comment:
I have not tested this suggestion, but the general idea is to create a class method  that generates dynamically  a sub class of pearsonBaseError so we could avoid the find_class_with_attribute_value implementation 
```suggestion
    @classmethod
    def from_dict(cls, data: dict):
        """Create an instance of Person or its subclass from a dictionary."""
        exception_type = data.get('exception_type')
        for subclass in cls.__subclasses__():
            if subclass.exception_type == exception_type:
                return subclass(**data)

        # Default to Person if no matching subclass is found
        return cls(**data )
```

Reply:
Applied in [c755803](https://github.com/eduNEXT/eox-nelp/pull/177/commits/c7558031226fd0629f990b03ebf202687844dff8)